NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -1679,9 +1679,9 @@ bool IsDefinedHashtypeSignature(byte[] vchSig)
 			}
 
 			var temp = ~(SigHash.AnyoneCanPay);
-			if((ScriptVerify & ScriptVerify.ForkId) != 0)

```

Comment:
No altcoin logic in NBitcoin code. ForkId is an exception for something multiple coin will use. If this does not handle your case then you need to refactor NBitcoin in a more general way which make very obvious to me that there is no change of behavior for what existed before.